File: fibonacci-demo/src/fibonacci.ts
export function calculateFibonacci(n: number): number {
    if (n < 0) {
        throw new Error("Input must be a non-negative integer.");
    }
    if (n === 0) return 0;
    if (n === 1) return 1;

    let a = 0, b = 1, temp: number;

    for (let i = 2; i <= n; i++) {
        temp = a + b;
        a = b;
        b = temp;
    }

    return b;
}

// Recursive implementation for demonstration
export function calculateFibonacciRecursive(n: number): number {
    if (n < 0) {
        throw new Error("Input must be a non-negative integer.");
    }
    if (n === 0) return 0;
    if (n === 1) return 1;

    return calculateFibonacciRecursive(n - 1) + calculateFibonacciRecursive(n - 2);
}

File: fibonacci-demo/Fibonacci.cs
// returns the nth Fibonacci number
public class FibonacciCalculator
{
    private const int BASE_THRESHOLD = 1;

    private static int FibRecursive(int n)
    {
        if (n <= BASE_THRESHOLD)
        {
            return n;
        }
        return FibRecursive(n - 1) + FibRecursive(n - 2);
    }

    public static int Fib(int n)
    {
       // Validate input
        if (n < 0)
            throw new ArgumentException("n must be non-negative", nameof(n));
        return FibRecursive(n);
    }
}
